cli/js/lib.deno.ns.d.ts

```diff
@@ -200,6 +214,10 @@ declare namespace Deno {
    *       console.log(myEnv.TEST_VAR === newEnv.TEST_VAR);  // outputs "true"
    *
    * Requires `allow-env` permission. */
+  // TODO(ry) Deno.env should be a magical proxy not function
+  // setting: Deno.env.path = "foo"
+  // getting: Deno.env.path
+  // get all: Deno.env
```

Comment:
@ry @piscisaureus I'm on the fence here - having explicit getters/setters is tempting - it'd be much easier to audit/lint access to environmental variables; otherwise you'd need to consider more cases:
```
Deno.env["foo"] = "asdf";
Deno.env.foo = "asdf"

vs.
Deno.env.get() // returns map
Deno.env.get("foo") // returns value for "foo"
Deno.env.set("foo", "asdf"); // sets value for foo

```

Reply:
I think your proposed API is better since it's more explicit. 

What about:
```js
Deno.envGet(); // everything
Deno.envGet("PATH"); // just $PATH
Deno.envSet("PATH", "/"); 
```

Reply:
Your solution works as well - if we're in agreement that making it least magical possible is desired then it boils down to preference between proposed solutions.

Reply:
To me this sounds like Object vs Map vs special ("magic") syntax.

I'd be in favor of using env like [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) since that would make it standardized automatically without special syntax.

```ts
Object.fromEntries(Deno.env) // all as object
Deno.env.get("PATH"); // just $PATH
Deno.env.set("PATH", "/");
```
This also would also come for free:
```ts
Deno.env.has("PATH"); // true or false
```

This way there would be no "magic" at all.

Reply:
@timreichen That's adding side-effects to methods of a built-in object. Pretty magical.

Reply:
What kind of side-effects do you mean?